sound/soc/sof/topology.c

```diff
@@ -2272,7 +2352,11 @@ int snd_sof_load_topology(struct snd_soc_component *scomp, const char *file)
 		return ret;
 	}
 
-	ret = snd_soc_tplg_component_load(scomp, &sof_tplg_ops, fw);
+	if (sof_debug_check_flag(SOF_DBG_DSPLESS_MODE))
+		ret = snd_soc_tplg_component_load(scomp, &sof_dspless_tplg_ops, fw);
+	else
+		ret = snd_soc_tplg_component_load(scomp, &sof_tplg_ops, fw);
+
```

Comment:
nit-pick: use a temp pointer and have a single call to tplg_component_load?

Reply:
I think this is clear enough, I don's see the benefit in this particular case.